src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -1219,4 +1221,82 @@ export class WorkspacesUsecases {
   }> {
     return this.teamRepository.getTeamUserAndTeamByTeamId(userUuid, teamId);
   }
+
+  async deleteWorkspaceContent(
+    workspaceId: Workspace['id'],
+    user: User,
+  ): Promise<void> {
+    const workspace = await this.workspaceRepository.findById(workspaceId);
+
+    if (!workspace) {
+      throw new NotFoundException('Workspace not found');
+    }
+
+    if (!workspace.isUserOwner(user)) {
+      throw new ForbiddenException('You are not the owner of this workspace');
+    }
+
+    const workspaceUser = await this.userRepository.findByUuid(
+      workspace.workspaceUserId,
+    );
+
+    const rootFolder = await this.folderUseCases.getByUuid(
+      workspace.rootFolderId,
+    );
+
+    await this.folderUseCases.deleteByUser(workspaceUser, [rootFolder]);
+
+    await this.workspaceRepository.deleteById(workspaceId);
+  }
+
+  async leaveWorkspace(
```

Comment:
There is one thing missing here: 
- All the items of that user in its personal workspace should be transferred to the owner's personal folder within that workspace.